src/components/vectorfeedback/VectorFeedbackDirective.js

```diff
@@ -60,9 +67,36 @@ goog.require('ga_translation_service');
 
         scope.$on('gaBgChange', function(evt, bg) {
           if (bg.disableEdit) {
+            scope.olLayer = null;
             toggle(false);
+          } else {
+            scope.olLayer = bg.olLayer;
           }
         });
+
+        scope.$watch('olLayer', function(newVal, oldVal) {
+          if (newVal && newVal !== oldVal) {
+            scope.styleUrls = gaLayers.getLayerProperty(scope.olLayer.id,
+                'styleUrls');
+          }
+        });
+
+        scope.applyNextStyle = function(styleUrl) {
+          var currUrl = scope.olLayer.externalStyleUrl;
+          var idx = scope.styleUrls.indexOf(currUrl);
+          if (!currUrl) {
+            idx = 1;
+          } else if (idx !== -1) {
+            idx++;
+            if (idx > scope.styleUrls.length) {
+              idx = 0;
+            }
+          } else {
+            idx = 0;
+          }
```

Comment:
that leads to a bug.
1. open edit mode
2. edit the style
3. click select another style

Result:
The map is reset

Expected result:
The next style is used